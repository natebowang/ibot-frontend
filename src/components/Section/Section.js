import React from "react";
import Radium from "radium";

import curvyLines from './curvyLines.png';
import {SECONDARY_COLOR} from "../../app/config";

const Section = ({children, bkImg = false, style}) => {
    return (
        <section style={bkImg ? {...sectionWithBkImgStyle, ...style} : {...sectionStyle, ...style}}>
            {children}
        </section>
    )
};

const sectionStyle = {
    flex: '1 0 auto'
};

const sectionWithBkImgStyle = {
    ...sectionStyle,
    backgroundColor: SECONDARY_COLOR,
    backgroundImage: `url(${curvyLines})`,
    backgroundRepeat: 'no-repeat',
    backgroundSize: '120% 200%',
    // backgroundSize: 'auto auto',
    // backgroundPositionX: 'center',
    backgroundPositionY: '-5rem',
};

export default Radium(Section);
