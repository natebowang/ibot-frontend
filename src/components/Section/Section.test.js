import React from "react";
import {create} from 'react-test-renderer';
import Section from "./Section";

test('Section without background image snapshot test', () => {
    const section = create(
        <Section/>
    );
    const tree = section.toJSON();
    expect(tree).toMatchSnapshot()
});

test('Section with background image snapshot test', () => {
    const section = create(
        <Section bkImg={true}/>
    );
    const tree = section.toJSON();
    expect(tree).toMatchSnapshot()
});
