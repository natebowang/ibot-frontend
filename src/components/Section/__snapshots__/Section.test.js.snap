// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section with background image snapshot test 1`] = `
<section
  data-radium={true}
  style={
    Object {
      "backgroundColor": "#FFFAEE",
      "backgroundImage": "url(curvyLines.png)",
      "backgroundPositionY": "-5rem",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "120% 200%",
      "flex": "1 0 auto",
    }
  }
/>
`;

exports[`Section without background image snapshot test 1`] = `
<section
  data-radium={true}
  style={
    Object {
      "flex": "1 0 auto",
    }
  }
/>
`;
